### PR TITLE
Adds modified syringe guns

### PR DESCRIPTION
**Pull request by swindly**
:cl: Swindly
add: Added modified syringe guns. They fire DNA injectors. Geneticists and CMOs can buy them from the traitor uplink for 14 TC.
/:cl:
I am not good at naming things.
[why]:  (Please add a short description [on the next line] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding:) 
A gun that shoots DNA injectors gives geneticists something to do besides greentide.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
"experimental syringe gun" may work.

**Comment by XDTM** [comment]
I think they have the potential to be extremely strong, even if they are temporary. No chem mix can replicate instant mute + blindness + deafness + random stuns.

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 1]
```diff
@@ -78,3 +78,29 @@
...
 	suppressed = 1 //Softer fire sound
 	can_unsuppress = 0 //Permanently silenced
+	
+/obj/item/weapon/gun/syringe/dna
+	name = "modified syringe gun"
+	desc = "A syringe gun that has been modified to fit DNA injectors instead of normal syringes."
+	origin_tech = "combat=2;syndicate=2;biotech=3"
+	
+/obj/item/weapon/gun/syringe/dna/New()
+	..()
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/syringe_gun.dm]: Initialize and . = ..()

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 2]
```diff
@@ -78,3 +78,29 @@
...
+	chambered = new /obj/item/ammo_casing/dnainjector(src)
+	
+/obj/item/weapon/gun/syringe/dna/attackby(obj/item/A, mob/user, params, show_msg = 1)
+	if(istype(A, /obj/item/weapon/dnainjector))
+		var/obj/item/weapon/dnainjector/D = A
+		if(D.used)
+			to_chat(usr, "<span class='warning'>This injector is used up!</span>")
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/syringe_gun.dm]: user

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 3]
```diff
@@ -78,3 +78,29 @@
...
+			return
+		if(syringes.len < max_syringes)
+			if(!user.transferItemToLoc(D, src))
+				return
+			to_chat(user, "<span class='notice'>You load [D] into \the [src].</span>")
+			syringes.Add(D)
+			recharge_newshot()
+			return 1
+		else
+			to_chat(usr, "<span class='warning'>[src] cannot hold more syringes!</span>")
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/syringe_gun.dm]: user

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 4]
```diff
@@ -78,3 +78,29 @@
...
+		if(D.used)
+			to_chat(usr, "<span class='warning'>This injector is used up!</span>")
+			return
+		if(syringes.len < max_syringes)
+			if(!user.transferItemToLoc(D, src))
+				return
+			to_chat(user, "<span class='notice'>You load [D] into \the [src].</span>")
+			syringes.Add(D)
+			recharge_newshot()
+			return 1
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/syringe_gun.dm]: TRUE

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 5]
```diff
@@ -78,3 +78,29 @@
...
+		else
+			to_chat(usr, "<span class='warning'>[src] cannot hold more syringes!</span>")
+	return 0
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/syringe_gun.dm]: FALSE

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 6]
```diff
@@ -78,3 +78,29 @@
...
+/obj/item/weapon/gun/syringe/dna
+	name = "modified syringe gun"
+	desc = "A syringe gun that has been modified to fit DNA injectors instead of normal syringes."
+	origin_tech = "combat=2;syndicate=2;biotech=3"
+	
+/obj/item/weapon/gun/syringe/dna/New()
+	..()
+	chambered = new /obj/item/ammo_casing/dnainjector(src)
+	
+/obj/item/weapon/gun/syringe/dna/attackby(obj/item/A, mob/user, params, show_msg = 1)
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/syringe_gun.dm]: TRUE

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 7]
```diff
@@ -78,3 +78,29 @@
...
+	if(istype(A, /obj/item/weapon/dnainjector))
+		var/obj/item/weapon/dnainjector/D = A
+		if(D.used)
+			to_chat(usr, "<span class='warning'>This injector is used up!</span>")
+			return
+		if(syringes.len < max_syringes)
+			if(!user.transferItemToLoc(D, src))
+				return
+			to_chat(user, "<span class='notice'>You load [D] into \the [src].</span>")
+			syringes.Add(D)
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/syringe_gun.dm]: +=

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 8]
```diff
@@ -78,3 +78,29 @@
...
+/obj/item/weapon/gun/syringe/dna/attackby(obj/item/A, mob/user, params, show_msg = 1)
+	if(istype(A, /obj/item/weapon/dnainjector))
+		var/obj/item/weapon/dnainjector/D = A
+		if(D.used)
+			to_chat(usr, "<span class='warning'>This injector is used up!</span>")
+			return
+		if(syringes.len < max_syringes)
+			if(!user.transferItemToLoc(D, src))
+				return
+			to_chat(user, "<span class='notice'>You load [D] into \the [src].</span>")
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/syringe_gun.dm]: \the [D]

File: code/modules/projectiles/guns/syringe_gun.dm [hunk 9]
```diff
@@ -78,3 +78,29 @@
...
+	
+/obj/item/weapon/gun/syringe/dna/attackby(obj/item/A, mob/user, params, show_msg = 1)
+	if(istype(A, /obj/item/weapon/dnainjector))
+		var/obj/item/weapon/dnainjector/D = A
+		if(D.used)
+			to_chat(usr, "<span class='warning'>This injector is used up!</span>")
+			return
+		if(syringes.len < max_syringes)
+			if(!user.transferItemToLoc(D, src))
+				return
```
Comment (Cyberboss) [inline code/modules/projectiles/guns/syringe_gun.dm]: return FALSE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile/bullets.dm [hunk 10]
```diff
@@ -240,7 +240,24 @@
...
+	var/obj/item/weapon/dnainjector/injector
+	
+/obj/item/projectile/bullet/dnainjector/on_hit(atom/target, blocked = 0)
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		if(blocked != 100)
+			if(M.can_inject(null, 0, def_zone, 0))
+				injector.inject(M, firer)
+				qdel(injector)
+				return 1
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: TRUE

File: code/modules/projectiles/projectile/bullets.dm [hunk 11]
```diff
@@ -240,7 +240,24 @@
...
+	icon_state = "syringeproj"
+	var/obj/item/weapon/dnainjector/injector
+	
+/obj/item/projectile/bullet/dnainjector/on_hit(atom/target, blocked = 0)
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		if(blocked != 100)
+			if(M.can_inject(null, 0, def_zone, 0))
+				injector.inject(M, firer)
+				qdel(injector)
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: QDEL_NULL

File: code/modules/projectiles/projectile/bullets.dm [hunk 12]
```diff
@@ -240,7 +240,24 @@
 		nodamage = 1
 	. = ..() // Execute the rest of the code.
 
-
+/obj/item/projectile/bullet/dnainjector
+	name = "\improper DNA injector"
+	icon_state = "syringeproj"
+	var/obj/item/weapon/dnainjector/injector
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: You need to QDEL_NULL this in Destroy

File: code/modules/projectiles/projectile/bullets.dm [hunk 13]
```diff
@@ -240,7 +240,24 @@
...
+	name = "\improper DNA injector"
+	icon_state = "syringeproj"
+	var/obj/item/weapon/dnainjector/injector
+	
+/obj/item/projectile/bullet/dnainjector/on_hit(atom/target, blocked = 0)
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		if(blocked != 100)
+			if(M.can_inject(null, 0, def_zone, 0))
+				injector.inject(M, firer)
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: You should also be checking the result of this cause you never know

File: code/modules/projectiles/projectile/bullets.dm [hunk 14]
```diff
@@ -240,7 +240,24 @@
...
+/obj/item/projectile/bullet/dnainjector
+	name = "\improper DNA injector"
+	icon_state = "syringeproj"
+	var/obj/item/weapon/dnainjector/injector
+	
+/obj/item/projectile/bullet/dnainjector/on_hit(atom/target, blocked = 0)
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		if(blocked != 100)
+			if(M.can_inject(null, 0, def_zone, 0))
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: 2nd and 4th parameters are bools, so FALSE

File: code/modules/projectiles/projectile/bullets.dm [hunk 15]
```diff
@@ -240,7 +240,24 @@
...
+				injector.inject(M, firer)
+				qdel(injector)
+				return 1
+			else
+				blocked = 100
+				target.visible_message("<span class='danger'>The [name] was deflected!</span>", \
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: \The [src]

File: code/modules/projectiles/projectile/bullets.dm [hunk 16]
```diff
@@ -240,7 +240,24 @@
...
+									   "<span class='userdanger'>You were protected against the [name]!</span>")
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: \the [src]

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Repukan** [comment]
Should be more expensive given that the ammo for it it's free

**Comment by Lumipharon** [comment]
This is hilarious(ly powerful).

**Comment by Cheridan** [comment]
ammo is free on every map but box because of public autolathe memes so i dont really think that's an issue

**Comment by Repukan** [comment]
>Get gloves
>Hack machine 
>Put metal
>Print ammo with no one looking

Equal to

>Press injector button on computer which nobody is going to question because it's your job.


Hmmmmmmmmmmmmmmmmmmmmmmmm

**Comment by Lumipharon** [comment]
Semi related question, but if an explosion embeds a syringe (of any sort) into someone, does it inject the contents into them, and if not, why?

**Comment by swindly** [comment]
>if an explosion embeds a syringe (of any sort) into someone, does it inject the contents into them

No because nobody implemented that behavior.

**Comment by octareenroon91** [comment]
Also thrown syringes injecting would make the syringe gun obsolete

**Comment by Cyberboss** [comment]
thrown != embed

File: code/modules/projectiles/projectile/bullets.dm [hunk 17]
```diff
@@ -240,7 +240,28 @@
...
+	name = "\improper DNA injector"
+	icon_state = "syringeproj"
+	var/obj/item/weapon/dnainjector/injector
+	
+/obj/item/projectile/bullet/dnainjector/on_hit(atom/target, blocked = 0)
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		if(blocked != 100)
+			if(M.can_inject(null, FALSE, def_zone, FALSE))
+				injector.inject(M, firer)
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: Check the result of this

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Comment by Cyberboss** [comment]
>i don't know what this does

M8 you wrote it https://github.com/tgstation/tgstation/pull/27276/files#diff-67a9e2657d08d6b449141b45ba81942bR53

**Comment by swindly** [comment]
The bullet is supposed to have the same behavior after injecting regardless of whether the target's DNA changed.

**Comment by kevinz000** [comment]
this looks like it's going to be rediculously op as no one wants to be ggnored by a monkey se syringe.

**Comment by Cyberboss** [comment]
Also I agree, this is definitely a 14 maybe 16+TC item

**Comment by Jazaen** [comment]
Yup, it's insta-monkey for everyone not wearing hardsuits and such.

Besides, while getting an SE with all negative mutations would take a while, all of them combined make someone worse than dead.
Deaf, Blind, Mute, with a chance to (tourettes) lose all control for a while.
As a note, I'm pretty sure changelings can also do that with transformation sting, but they have to extract DNA beforehand (or it's lost), can do only two in rapid succession due to prohibitive chemical cost, and need to stand right next to the target.

**Comment by calzilla1** [comment]
Isn't this just a replacement to mutagen syringe?

**Comment by Jazaen** [comment]
I'd disagree. Mutagen can cripple someone, sure, but it takes a little RnG for that to happen quickly. When you monkey someone, they instantly lose all of their gear.

**Comment by Lumipharon** [comment]
injecting mutagen isn't that powerful. Sure, it's a fairly strong toxin, but it has low chances to proc mutations, and it's entirely random.
Mutagen smoke is/was powerful because the smoke would cause the mutation check thing like 5000 times in 1 second, so you're just instantly fucked.

**Comment by ChangelingRain** [comment]
Should probably be 12 TC, because it's about as powerful as a revolver. It's basically a fancy revolver.

**Comment by AdamElTablawy** [comment]
Eh, revolvers kill. This debilitates, but doesn't kill by itself, and wears off. 

On the other hand, revolver hits are softened by bulletproof armor. This isn't softened by anything.

I can't see it being used to murderbone like the revolver by itself. Maybe piss people off a lot. Make it easier to kill them. But not kill by itself. Plus, mutadone counters it fairly quickly.

Maybe 10 TC? 10-12ish seems fair, since it's a one shot debilitate/monkey but doesn't kill by itself and can be cockblocked by mutadone.

**Comment by Repukan** [comment]
>Stack Blindess, deafness, muteness, epilepsy, and monkey

>Stab and butcher as you go

Free ammo on tap too.

**Comment by octareenroon91** [comment]
>Mutadone is a counter

So the traitor geneticist dumps the mutadone in their office leaving you to wait for chemistry to remember they have a job, and then the recipe for mutadone.

**Comment by Lumipharon** [comment]
revolver is double tap crit, 6 rounds a pop.
A syringe gun is a single syringe which will horribly cripple you/monkey you.

Revolver is stopped by armour - but most armour commonly used won't stop the double tap (need hos tier/bullet proof etc).
Syringe is stopped by pierce immunity (hardsuit/biosuit/being a borg etc)

Revolver is loud and angry, but needs a autolathe for more ammo.
Syringe gun needs just the dna machine (which means it's not obvious that they're a traitor, but probs a bit slower to mass produce loads of syringes.

Overall revolver is more deadly and useful - you can juke a few revolver shots and still get rekt before they have to reload - with a syringe gun having to reload after each shot means it's harder a weapon of mass destruction.

**Comment by Jazaen** [comment]
Pretty sure you can't stack negative mutations with monkey SE, in that case, because it only applies them once, and monkey transformation has precedence over negative SEs.
Never tested that one though.

**Comment by jughu** [comment]
Just give them a decloner in that kind of sense xd

**Comment by lzimann** [comment]
feel like shooting dna injectors is going to be strong as fuck
@KorPhaeron

**Pull request merged**